Add an Affiliated Packages page

rocco8773 opened the pull request:
This PR creates the "Affiliated Packages" page and defines the framework for adding community/affiliated packages in the future.

What it does?

1. Removes the "Page Source" link from the website altogether.
1. Moves the "Benchmarks" page link to the footer.
1. Adds the "Affiliated Packages" link to the top menu.
1. Created a Google Form so projects/organizations can submit a request to be added to the affiliated packages.
1. Framework for building the "Affiliated Packages" page:
    * Affiliated package information is stored in a json file, `/web/data/affiliated_pkgs.json`.
    * Several `Mako` style shortcodes (`/web/shortcodes/`) and template (`/web/templates/`) templates were created to automate the html generation of the list.
    * The affiliated packages list is automatically generated using the shortcodes and the json file, so adding a new entry to the json file will automatically add the package to the page upon rebuild.
    * Created a shield.io badge to distiguish PlasmaPy affiliated packages from all other community packages.
    * Generated list is automatically alphabetized.
1. Added CSS styling to make it all look pretty
1. Updated the chat links in the top menu and footer to point to Element, not Riot.

**Note:**  This PR should not be merged before we gather a few community packages.  I'm assuming we'll have community packages after the APS followup meetings next week.  I just wanted this ready for merge so we can advertise it at the followup meetings.

----

### NEW

<img width="713" alt="image" src="https://user-images.githubusercontent.com/29869348/99125276-904b5b80-25b8-11eb-9a3c-f638de8200ae.png">


### OLD

![image](https://user-images.githubusercontent.com/29869348/99125367-c12b9080-25b8-11eb-80e4-dbc784c58c7e.png)

StanczakDominik commented:
Built version shows up at https://179-105815475-gh.circle-artifacts.com/0/web/output/affiliates/index.html

StanczakDominik commented on web/data/affiliated_pkgs.json:
```diff
@@ -0,0 +1,48 @@
+      "short": "A open-source Python 3 package for plasma physics.",
+      "long": "PlasmaPy is an open-source community developed Python 3 package for plasma physics.  PlasmaPy intends to be a collection of functionality commonly used and shared between plasma physicists and researchers globally."
+    },
+    "affiliated": true,
```
`RecursionError`? :smiley: 
```suggestion
    "affiliated": false,
```

StanczakDominik commented on web/data/affiliated_pkgs.json:
```diff
@@ -0,0 +1,48 @@
+{
```
This whole concept of plugging in a json file is pretty badass, kudos!

StanczakDominik commented on web/data/affiliated_pkgs.json:
```diff
@@ -0,0 +1,48 @@
+    "status": "perpetual beta",
+    "license": "MIT"
+  }
+}
```
Do we want to add https://github.com/PlasmaPy/PlasmaPy-NEI/ in here?

rocco8773 replied:
@namurphy you ready for PlasmaPy-NEI to be in this?  Essentially, I wasn't going to add any package until the maintainers requested it to be added.

namurphy replied:
I'd say it should be added after there's a 0.1 release, so it doesn't need to be included yet.

StanczakDominik commented on web/data/affiliated_pkgs.json:
```diff
@@ -0,0 +1,48 @@
+    "status": "perpetual beta",
+    "license": "BSD 3-clause with patent grant"
+  },
+  "tofu": {
```
I need to finally catch up on the tofu stuff... I just know it goes well with chills.

rocco8773 replied:
Yeah, next week is going to catch up on everything dripped over the last two weeks.

StanczakDominik reviewed (commented):
That shortcode stuff looks really cool! What resources did you use to figure it out?

LGTM with a few comments - like you said, let's wait with the merge :)

rocco8773 commented:
> That shortcode stuff looks really cool! What resources did you use to figure it out?

I discovered [shortcodes in the Nikola documentation](https://getnikola.com/handbook.html#shortcodes) when @namurphy mentioned automating the meeting times in PR #79.  I have to say though, THE DOCUMENTATION ON THIS IS ATROCIOUS!!!!  The format used for shortcodes is either `mako` or `jinja2`, depending on the theme your using.  `bootstrap4` (our theme) is `mako`.  Additionally,  if you want to make shortcodes that are importable by other shortecodes, then that file must be placed in the `/web/templates/` folder.  This took me hours to figure out from this terribly written sentence in the Nikola documentation...

> For example, if you define a macro foo(x, y) in templates/shared_sc.tmpl, you can include shared_foo.tmpl in templates/special_post.tmpl and shortcodes/foo.tmpl and then call the ${shared_foo.foo(x, y)} macro.

All grumbling aside, this is a very nifty tool to have once you understand how it works.

rocco8773 commented:
Thought...we should make "Affiliated" and "Community" shields that could be copied from the website and added to other project's websites and/or package's README's.

StanczakDominik commented on web/pages/affiliates/index.md:
```diff
@@ -19,9 +19,10 @@ submit a request.
-`plasmapy` interoperability are marked as **Affiliated Packages**. 
-{{% affiliated_shield %}}
+All projects/packages are considered **Community Packages** 
+<span style="vertical-align: 2px">{{% community_shield %}}</span>, but packages design with
```
```suggestion
<span style="vertical-align: 2px">{{% community_shield %}}</span>, but packages designed with
```

StanczakDominik commented on web/templates/shields.tmpl:
```diff
@@ -1,4 +1,21 @@
+## shield.io's used for PlasmaPy
+
+<%def name="affiliated_src()">
+    ##  source url for the affiliated PlsamaPy shield.io
```
```suggestion
    ##  source url for the affiliated PlasmaPy shield.io
```

rocco8773 commented:
Ok, I've added several styling changes to the the display...

1. If a project does not provide an image, then their name will be automatically placed into the pkg thumbnail.
1. I added a field for the pkg license.
1. The pkg description, license, and keyword fields will automatically scroll vertically on overflow.

The current build (at the time of this comment) can be found [here](https://209-105815475-gh.circle-artifacts.com/0/web/output/affiliates/index.html).

@namurphy Can you please look over the PlasmaPy entry and give any comments on the text?

@Didou09 Can you look over the `tofu` section and give any comments or approval?

Didou09 commented:
Good for me @rocco8773 , nice page :+1: 
Question: how do I get the nice 
![image](https://user-images.githubusercontent.com/5929562/101379162-43575e00-38b4-11eb-806a-f24163ee6468.png)
 badge to put it on tofu's page too ?

rocco8773 commented:
@Didou09 Thank you!

Good question! I plan to put instructions on how to add a shield to your repo in a follow up PR, but for now you can add the the shield by doing the following...

1. [copy this link address](
https://img.shields.io/badge/Affiliated-PlasmaPy-ff2929?labelColor=435466&logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAAAAXNSR0IArs4c6QAAAIRlWElmTU0AKgAAAAgABQESAAMAAAABAAEAAAEaAAUAAAABAAAASgEbAAUAAAABAAAAUgEoAAMAAAABAAIAAIdpAAQAAAABAAAAWgAAAAAAAADIAAAAAQAAAMgAAAABAAOgAQADAAAAAQABAACgAgAEAAAAAQAAAA6gAwAEAAAAAQAAAA4AAAAAnETOywAAAAlwSFlzAAAewgAAHsIBbtB1PgAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAAAjdJREFUKBVdkk9IVFEUxs99f6bXjDNS4TRvMiHQImsRpBCZEC5clLOKZtFOIdrmzmWroM0IuotAcOlshlwpGEFkuBAM+ycFiomODE6O49x58+a+e/vu2MjQg998591zzj33fvOI/j2KyGyJjWastTXXun6aQMEtMAOWwTvwUoXDSV2cyWTO0tTUmdNGJC39An0CDkAJ7IJ94EvG1qEJWliIP5ydvU5KMV1v4FcgcRfxNCiCTVAAR2Sab5lSr2Qi8Ywcp90Soqszm22coHmX541CTIT6oEaW9ZPa2uqIx467uy/S4WHcC4WunCe6jTWyMM2FdoE84Mo0TYrHi2xvr0Clkl4vrfX350jKvorj9OAEEawRQ+M16GtQVpZFzDA4+f4S3u3iyIi5Ojj4cbi3d2hAiHtcSo7G3ZqUE9oYDxTItn1Wr1cRz9xYXHSVlMPfHaeMTfoGOL/JcQ3GWASTRUelYujGI7AvXFdY29sfxnK5oQrRi06l1u5Xqwe+EK7HGMHEdjjqKMPYeT866mlX/5Bh/EDTLxhS3gyFHrhK5QS2L0vZUyO6QIxFUReBXoKbKxhEJ65KuRxEo8ckxO8ty1rBPdalUkkUn0Ox/qJM6FXs9c0Lgje6sfHnQ7+IZLLD3NjIb9n2dMLzPpuxWFhyTrgrLGQUcP7JZuzR13Tafzw3Z2pXsYzTE8V2xsfrlycnq6n5+XDetp+i6Q7MQFItraZSjUm6KZtOB3qq/tzQe/LoRDP+X1tzfwGubPWENrTteQAAAABJRU5ErkJggg==)
2. put the following markdown snippet into your `README.md` `[![PlamsaPy](paste link address)](https://www.plasmapy.org/affiliates/)`
3. this should render like [![PlasmaPy](https://img.shields.io/badge/Affiliated-PlasmaPy-ff2929?labelColor=435466&logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAAAAXNSR0IArs4c6QAAAIRlWElmTU0AKgAAAAgABQESAAMAAAABAAEAAAEaAAUAAAABAAAASgEbAAUAAAABAAAAUgEoAAMAAAABAAIAAIdpAAQAAAABAAAAWgAAAAAAAADIAAAAAQAAAMgAAAABAAOgAQADAAAAAQABAACgAgAEAAAAAQAAAA6gAwAEAAAAAQAAAA4AAAAAnETOywAAAAlwSFlzAAAewgAAHsIBbtB1PgAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAAAjdJREFUKBVdkk9IVFEUxs99f6bXjDNS4TRvMiHQImsRpBCZEC5clLOKZtFOIdrmzmWroM0IuotAcOlshlwpGEFkuBAM+ycFiomODE6O49x58+a+e/vu2MjQg998591zzj33fvOI/j2KyGyJjWastTXXun6aQMEtMAOWwTvwUoXDSV2cyWTO0tTUmdNGJC39An0CDkAJ7IJ94EvG1qEJWliIP5ydvU5KMV1v4FcgcRfxNCiCTVAAR2Sab5lSr2Qi8Ywcp90Soqszm22coHmX541CTIT6oEaW9ZPa2uqIx467uy/S4WHcC4WunCe6jTWyMM2FdoE84Mo0TYrHi2xvr0Clkl4vrfX350jKvorj9OAEEawRQ+M16GtQVpZFzDA4+f4S3u3iyIi5Ojj4cbi3d2hAiHtcSo7G3ZqUE9oYDxTItn1Wr1cRz9xYXHSVlMPfHaeMTfoGOL/JcQ3GWASTRUelYujGI7AvXFdY29sfxnK5oQrRi06l1u5Xqwe+EK7HGMHEdjjqKMPYeT866mlX/5Bh/EDTLxhS3gyFHrhK5QS2L0vZUyO6QIxFUReBXoKbKxhEJ65KuRxEo8ckxO8ty1rBPdalUkkUn0Ox/qJM6FXs9c0Lgje6sfHnQ7+IZLLD3NjIb9n2dMLzPpuxWFhyTrgrLGQUcP7JZuzR13Tafzw3Z2pXsYzTE8V2xsfrlycnq6n5+XDetp+i6Q7MQFItraZSjUm6KZtOB3qq/tzQe/LoRDP+X1tzfwGubPWENrTteQAAAABJRU5ErkJggg==)](https://www.plasmapy.org/affiliates/)

**Note:** The shield link will return a "page not found" until this PR is merged.

For the Community Shield, you'd [copy this link address](
https://img.shields.io/badge/Community-PlasmaPy-54bfbf?labelColor=435466&logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAAAAXNSR0IArs4c6QAAAIRlWElmTU0AKgAAAAgABQESAAMAAAABAAEAAAEaAAUAAAABAAAASgEbAAUAAAABAAAAUgEoAAMAAAABAAIAAIdpAAQAAAABAAAAWgAAAAAAAADIAAAAAQAAAMgAAAABAAOgAQADAAAAAQABAACgAgAEAAAAAQAAAA6gAwAEAAAAAQAAAA4AAAAAnETOywAAAAlwSFlzAAAewgAAHsIBbtB1PgAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAAAjdJREFUKBVdkk9IVFEUxs99f6bXjDNS4TRvMiHQImsRpBCZEC5clLOKZtFOIdrmzmWroM0IuotAcOlshlwpGEFkuBAM+ycFiomODE6O49x58+a+e/vu2MjQg998591zzj33fvOI/j2KyGyJjWastTXXun6aQMEtMAOWwTvwUoXDSV2cyWTO0tTUmdNGJC39An0CDkAJ7IJ94EvG1qEJWliIP5ydvU5KMV1v4FcgcRfxNCiCTVAAR2Sab5lSr2Qi8Ywcp90Soqszm22coHmX541CTIT6oEaW9ZPa2uqIx467uy/S4WHcC4WunCe6jTWyMM2FdoE84Mo0TYrHi2xvr0Clkl4vrfX350jKvorj9OAEEawRQ+M16GtQVpZFzDA4+f4S3u3iyIi5Ojj4cbi3d2hAiHtcSo7G3ZqUE9oYDxTItn1Wr1cRz9xYXHSVlMPfHaeMTfoGOL/JcQ3GWASTRUelYujGI7AvXFdY29sfxnK5oQrRi06l1u5Xqwe+EK7HGMHEdjjqKMPYeT866mlX/5Bh/EDTLxhS3gyFHrhK5QS2L0vZUyO6QIxFUReBXoKbKxhEJ65KuRxEo8ckxO8ty1rBPdalUkkUn0Ox/qJM6FXs9c0Lgje6sfHnQ7+IZLLD3NjIb9n2dMLzPpuxWFhyTrgrLGQUcP7JZuzR13Tafzw3Z2pXsYzTE8V2xsfrlycnq6n5+XDetp+i6Q7MQFItraZSjUm6KZtOB3qq/tzQe/LoRDP+X1tzfwGubPWENrTteQAAAABJRU5ErkJggg==).  Then you'd same markdown snippet as above to get [![PlasmaPy](
https://img.shields.io/badge/Community-PlasmaPy-54bfbf?labelColor=435466&logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAAAAXNSR0IArs4c6QAAAIRlWElmTU0AKgAAAAgABQESAAMAAAABAAEAAAEaAAUAAAABAAAASgEbAAUAAAABAAAAUgEoAAMAAAABAAIAAIdpAAQAAAABAAAAWgAAAAAAAADIAAAAAQAAAMgAAAABAAOgAQADAAAAAQABAACgAgAEAAAAAQAAAA6gAwAEAAAAAQAAAA4AAAAAnETOywAAAAlwSFlzAAAewgAAHsIBbtB1PgAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAAAjdJREFUKBVdkk9IVFEUxs99f6bXjDNS4TRvMiHQImsRpBCZEC5clLOKZtFOIdrmzmWroM0IuotAcOlshlwpGEFkuBAM+ycFiomODE6O49x58+a+e/vu2MjQg998591zzj33fvOI/j2KyGyJjWastTXXun6aQMEtMAOWwTvwUoXDSV2cyWTO0tTUmdNGJC39An0CDkAJ7IJ94EvG1qEJWliIP5ydvU5KMV1v4FcgcRfxNCiCTVAAR2Sab5lSr2Qi8Ywcp90Soqszm22coHmX541CTIT6oEaW9ZPa2uqIx467uy/S4WHcC4WunCe6jTWyMM2FdoE84Mo0TYrHi2xvr0Clkl4vrfX350jKvorj9OAEEawRQ+M16GtQVpZFzDA4+f4S3u3iyIi5Ojj4cbi3d2hAiHtcSo7G3ZqUE9oYDxTItn1Wr1cRz9xYXHSVlMPfHaeMTfoGOL/JcQ3GWASTRUelYujGI7AvXFdY29sfxnK5oQrRi06l1u5Xqwe+EK7HGMHEdjjqKMPYeT866mlX/5Bh/EDTLxhS3gyFHrhK5QS2L0vZUyO6QIxFUReBXoKbKxhEJ65KuRxEo8ckxO8ty1rBPdalUkkUn0Ox/qJM6FXs9c0Lgje6sfHnQ7+IZLLD3NjIb9n2dMLzPpuxWFhyTrgrLGQUcP7JZuzR13Tafzw3Z2pXsYzTE8V2xsfrlycnq6n5+XDetp+i6Q7MQFItraZSjUm6KZtOB3qq/tzQe/LoRDP+X1tzfwGubPWENrTteQAAAABJRU5ErkJggg==)](https://www.plasmapy.org/affiliates/).

I hope to make this process more efficient once I add the instruction to the website.  I'll ping you when this happens.

namurphy commented on web/conf.py:
```diff
@@ -176,21 +174,19 @@
                 (TWITTER, "Twitter"),
                 (YOUTUBE_CHANNEL, "YouTube"),
-                # (TELECON_NOTES, "Notes from weekly meetups"),
                 # (RITIEK, "GSoC 2018"),
```
```suggestion
```
Since it's a few years after 2018.  I hope this doesn't count as scope creep! :scream_cat:

namurphy commented on web/pages/affiliates/index.md:
```diff
@@ -0,0 +1,28 @@
+hidetitle: True
+author: Erik T. Everson
+
+The PlasmaPy's charter includes promoting open-source software for all of 
+the plasma community, and not just `plasmapy`.  We feel the entire community 
```
```suggestion
The PlasmaPy Project's charter includes promoting open-source software for all of 
the plasma community, and not just PlasmaPy.  We feel the entire community 
```
I tend to use "`plasmapy`" when talking specifically about code, and "PlasmaPy" as the proper noun.

rocco8773 replied:
I was referring to the package in this case.  This page contains a list of software packages and, hence, we're promoting packages so I was referring to `plasmapy`.

namurphy replied:
I was thinking we could follow how [Astropy](https://www.astropy.org/) does it...  They use "`astropy`" in snippets of code, like `import astropy`...and instead write out "Astropy" when talking about it as the name of a package.

rocco8773 replied:
Idk, to me having PlasmaPy refer to the project and PlasmaPy/`plasmapy` refer to the package is confusing.  In my mind things are a lot clearer if PlasmaPy is the overarching project and `plasmapy` is the package.  However, your the PlasmaPy Godfather so I'll defer to you.

namurphy commented on web/pages/affiliates/index.md:
```diff
@@ -0,0 +1,28 @@
+
+All projects/packages are considered **Community Packages** 
+<span style="vertical-align: 2px">{{% community_shield %}}</span>, but packages designed with
+`plasmapy` interoperability are marked as **Affiliated Packages** 
```
```suggestion
PlasmaPy interoperability are marked as **Affiliated Packages** 
```

rocco8773 replied:
I'm still think package here instead of project because "Affiliated Packages" have interoperability with `plasmapy`, not PlasmaPy.

namurphy reviewed (approved):
Looks good to me!  I only have a few minor suggested changes.
